velox/functions/lib/aggregates/SingleValueAccumulator.h

```diff
@@ -50,6 +50,12 @@ struct SingleValueAccumulator {
   /// method.
   int32_t compare(const DecodedVector& decoded, vector_size_t index) const;
 
+  static int32_t compare(
```

Comment:
Let's keep SingleValueAccumulator as a generic utility and not add Presto-specific behavior here.

To avoid copy-pasting related logic into multiple aggregate functions, let's move it to a separate .h / .cpp file. Maybe, 

```
prestosql/aggregates/Compare.h

int32_t compare(
      const SingleValueAccumulator* accumulator,
      const DecodedVector& decoded,
      vector_size_t index);
```

We can also remove CompareFlags argument since all Presto functions would use the same values.

Reply:
>Let's keep SingleValueAccumulator as a generic utility and not add Presto-specific behavior here.

Ok got it. I will find a suitable place.

>We can also remove CompareFlags argument since all Presto functions would use the same values.

It is used not only by the Presto functions but also the Spark functions, which has different null handling behavior as I described in the PR's description.

Reply:
> It is used not only by the Presto functions but also the Spark functions, which has different null handling behavior as I described in the PR's description. We need the flag to control the null mode.

This is correct. What I was trying to say is that Presto-specific function doesn't need to take CompareFlags.